bundles/org.openhab.core.config.discovery/src/main/java/org/openhab/core/config/discovery/AbstractDiscoveryService.java

```diff
@@ -454,4 +439,49 @@ private boolean getAutoDiscoveryEnabled(Object autoDiscoveryEnabled) {
     private String inferKey(DiscoveryResult discoveryResult, String lastSegment) {
         return "discovery." + discoveryResult.getThingUID().getAsString().replaceAll(":", ".") + "." + lastSegment;
     }
+
+    protected DiscoveryResult getLocalizedDiscoveryResult(final DiscoveryResult discoveryResult,
+            @Nullable Bundle bundle) {
+        if (i18nProvider != null && localeProvider != null && bundle != null) {
+            String defaultLabel = discoveryResult.getLabel();
+
+            String key = I18nUtil.stripConstantOr(defaultLabel, () -> inferKey(discoveryResult, "label"));
+
+            ParsedKey parsedKey = new ParsedKey(key);
+
+            String label = i18nProvider.getText(bundle, parsedKey.key, defaultLabel, localeProvider.getLocale(),
+                    parsedKey.args);
+
+            return DiscoveryResultBuilder.create(discoveryResult.getThingUID())
```

Comment:
Only a minor thought: Should we check if the label has changed before rebuilding a new discovery result? May safe some resources though.

Reply:
Yes, good idea.

Reply:
Fixed